### PR TITLE
🎨🐛 Replace template reference with correct details

### PR DESCRIPTION
**Pull request by astroash**
**Changes:**
- add query for detail templates
- add concept of templateName
- merge details and templates where needed
- some tests

### CODE REVIEW

File: src/test/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailServiceTest.kt [hunk 1]
```diff
@@ -135,11 +148,13 @@ internal class DetailServiceTest {
             val details = listOf(getValidShiftDetail(123L, 86400L))
             every { sqlRepository.getDetails(from, to, quantumId) } returns details
             every { authenticationFacade.currentUsername } returns quantumId
+            every { sqlRepository.getTemplateDetails(emptyList()) } returns emptyList<TemplateDetail>()
 
             val returnValue = service.getStaffDetails(from, to, quantumId)
 
+
             verify { sqlRepository.getDetails(from, to, quantumId) }
-            confirmVerified(sqlRepository)
```
Comment (astroash) [inline src/test/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailServiceTest.kt:142]: removed as this is run on `@AfterEach` hook

File: src/test/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailServiceTest.kt [hunk 2]
```diff
@@ -150,11 +247,12 @@ internal class DetailServiceTest {
             val details = listOf(getValidShiftDetail(86401L, 456L))
             every { sqlRepository.getDetails(from, to, quantumId) } returns details
             every { authenticationFacade.currentUsername } returns quantumId
+            every { sqlRepository.getDetailTemplates(emptyList()) } returns emptyList<DetailTemplate>()
 
             val returnValue = service.getStaffDetails(from, to, quantumId)
 
             verify { sqlRepository.getDetails(from, to, quantumId) }
-            confirmVerified(sqlRepository)
+            verify { sqlRepository.getDetailTemplates(emptyList()) }
```
Comment (nickmcmahon01) [inline src/test/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailServiceTest.kt:255]: Its still useful to make both of these assertions. confirmVerified means "and nothing else happened"

**Review by nickmcmahon01 — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailServiceTest.kt (hunk 2 above)
Comment (astroash) [reply src/test/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailServiceTest.kt:255]: See comment https://github.com/ministryofjustice/csr-api/pull/22#discussion_r488587005

File: src/main/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailService.kt [hunk 3]
```diff
@@ -15,14 +16,26 @@ class DetailService(private val sqlRepository: SqlRepository, val authentication
...
+        log.info("Found ${mergedDetails.size} shift details for $quantumId")
+
+        return mapToDetailsDto(mergedDetails)
     }
 
     fun getModifiedDetailsByPlanUnit(planUnit: String): Collection<DetailDto> {
         log.debug("Fetching modified shifts for $planUnit")
         val modifiedShifts = sqlRepository.getModifiedShifts(planUnit)
+
+        //TODO: add in template data
```
Comment (astroash) [inline src/main/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailService.kt:38]: @nickmcmahon01 do we wanna do this? I think we'd need it for notifications, but you've probably got a better idea of how that side fits together

Comment (nickmcmahon01) [reply src/main/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailService.kt:38]: I'm not sure, and to be blunt I'm not sure the team would know either. lets ignore it for now as its missing now in production (any currently doesn't use the plugin which this work replaces). I'll look into it.

**Review by nickmcmahon01 — APPROVED** [review]

**Pull request merged**